ui/js/controllers/perf/compare.js

```diff
@@ -188,35 +208,40 @@ perf.controller('CompareResultsCtrl', [
-                            originalRevision: $scope.originalRevision,
-                            newProject: $scope.newProject.name,
-                            newRevision: $scope.newRevision,
-                            originalSignature: oldSig,
-                            newSignature: newSig,
-                            framework: $scope.filterOptions.framework.id
-                        });
+                    if($scope.originalRevision !== undefined){
+                        if (testName.indexOf("summary") > 0) {
+                            var detailsLink = 'perf.html#/comparesubtest?';
+                            detailsLink += $httpParamSerializer({
+                                originalProject: $scope.originalProject.name,
+                                originalRevision: $scope.originalRevision,
+                                newProject: $scope.newProject.name,
+                                newRevision: $scope.newRevision,
+                                originalSignature: oldSig,
+                                newSignature: newSig,
+                                framework: $scope.filterOptions.framework.id
+                            });
+                            cmap.links.push({
+                                title: 'subtests',
+                                href: detailsLink
+                            });
+                        }
+
                         cmap.links.push({
-                            title: 'subtests',
-                            href: detailsLink
+                            title: 'graph',
+                            href: PhCompare.getGraphsLink(_.map(_.uniq(
+                                [$scope.originalProject, $scope.newProject]), function(project) {
+                                return {
+                                    projectName: project.name,
+                                    signature: oldSig,
+                                    frameworkId: $scope.filterOptions.framework.id
+                                };
+                            }), [$scope.originalResultSet,
+                                $scope.newResultSet])
                         });
                     }
-
-                    cmap.links.push({
-                        title: 'graph',
-                        href: PhCompare.getGraphsLink(_.map(_.uniq(
-                            [$scope.originalProject, $scope.newProject]), function(project) {
-                            return {
-                                projectName: project.name,
-                                signature: oldSig,
-                                frameworkId: $scope.filterOptions.framework.id
-                            };
-                        }), [$scope.originalResultSet,
-                             $scope.newResultSet])
-                    });
+                    else{
+                        //ShrutiJ: Graph and subtest links for baseline comparison
```

Comment:
Let's leave these blank for now, once you have something working I can some advice on how to add these features in. :)